src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -42,11 +43,23 @@ public KustoClient(string connectionString, string azureAccountToken, string own
         {
             _ownerUri = ownerUri;
             ClusterName = GetClusterName(connectionString);
-            DatabaseName = new SqlConnectionStringBuilder(connectionString).InitialCatalog;
+            
+            var dbName = new SqlConnectionStringBuilder(connectionString).InitialCatalog;
+            SetDatabaseName(dbName);
+            
             Initialize(ClusterName, DatabaseName, azureAccountToken);
             SchemaState = LoadSchemaState();
         }
 
+        private void SetDatabaseName(string databaseName)
+        {
+            var regex = new Regex(@"(?<=\().+?(?=\))");
+            
+            DatabaseName = regex.IsMatch(databaseName)
+                ? regex.Match(databaseName).Value
+                : databaseName;
+        }
+
```

Comment:
If the databaseName has `( )` then we want the real database name which is listed in the parenthesis, otherwise we can use the normal database name.